JDBC/DatabasesPractice/src/main/java/com/laba/solvd/database/DAO/ProfessorDAO.java

```diff
@@ -0,0 +1,167 @@
+package com.laba.solvd.database.DAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import com.laba.solvd.database.Model.Professor;
+import com.laba.solvd.database.Model.ResearchLab;
+import com.laba.solvd.database.Model.Student;
+import com.laba.solvd.database.Model.UniClass;
+
+public class ProfessorDAO implements IDAO<Professor>, ICreatableWithRelationship<Professor, UniClass> {
+	
+	@Override
+	public void create(Professor professor, UniClass uniClass) {
+		try (Connection connection = (Connection) ConnectionManager.getSQLConnection().get();) {

```

Comment:
You are creating connection in each DAO class , can you find an alternate approach to avoid such code duplication?